<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Supplementary Methods for Kukurba, et al. &mdash; ASE 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="ASE 1.0 documentation" href="#" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="#">ASE 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="#">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Supplementary Methods for Kukurba, et al.</a></li>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#part-1-aligning-expression-data"><strong>Part 1: Aligning Expression Data</strong></a><ul>
<li><a class="reference internal" href="#input-files">Input files</a></li>
<li><a class="reference internal" href="#align-fastq-files-using-star"><strong>1. Align fastq files using STAR</strong></a></li>
<li><a class="reference internal" href="#convert-format-of-expression-file"><strong>2. Convert format of expression file</strong></a></li>
</ul>
</li>
<li><a class="reference internal" href="#part-2-detecting-allele-specific-expression"><strong>Part 2: Detecting Allele-Specific Expression</strong></a><ul>
<li><a class="reference internal" href="#id1">Input files</a></li>
<li><a class="reference internal" href="#sort-alignments-using-samtools"><strong>1. Sort alignments using SAMtools</strong></a></li>
<li><a class="reference internal" href="#call-variants-using-samtools"><strong>2. Call variants using SAMtools</strong></a></li>
<li><a class="reference internal" href="#parse-pileup-file"><strong>3. Parse pileup file</strong></a></li>
<li><a class="reference internal" href="#create-gene-snp-mapping-file"><strong>4. Create gene-snp mapping file</strong></a></li>
<li><a class="reference internal" href="#binomial-test-for-ase"><strong>5. Binomial test for ASE</strong></a></li>
</ul>
</li>
<li><a class="reference internal" href="#additional-resources">Additional Resources</a></li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/index.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="toctree-wrapper compound">
<ul class="simple">
</ul>
</div>
<div class="section" id="supplementary-methods-for-kukurba-et-al">
<h1>Supplementary Methods for Kukurba, et al.<a class="headerlink" href="#supplementary-methods-for-kukurba-et-al" title="Permalink to this headline">¶</a></h1>
<p>This site provides the code and files necessary to replicate allele-specific expression (ASE) detection from mmPCR-Seq data published in the Kukurba, et al. paper. The steps outlined below shows how to call variants from expression data and test for ASE to identify loci with allelic imbalance at the expression level. The final output (ase.txt file) is equivalent to the processed files available on the Gene Expression Omnibus (GSE51769). These steps can be applied to to other expression data sets.</p>
</div>
<div class="section" id="dependencies">
<h1>Dependencies<a class="headerlink" href="#dependencies" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference external" href="https://code.google.com/p/rna-star/">STAR version 2.14 or newer</a> must be installed and be in your $PATH</li>
<li><a class="reference external" href="http://sourceforge.net/projects/samtools/files/tabix/">Tabix</a> must be installed and be in your $PATH</li>
<li><a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a> must be installed and be in your $PATH</li>
<li><a class="reference external" href="http://www.perl.org">Perl version 4.0 or newer</a></li>
<li><a class="reference external" href="http://www.r-project.org">R version 2.10 or newer</a></li>
<li><a class="reference external" href="http://www.gencodegenes.org/releases/">Gencode GTF annotation file</a></li>
</ul>
</div>
<div class="section" id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h1>
<ol class="arabic">
<li><p class="first">Install the above dependencies.</p>
</li>
<li><p class="first">Download the <a class="reference download internal" href="_downloads/samase.pl"><tt class="xref download docutils literal"><span class="pre">samase.pl</span></tt></a> script to the desired directory</p>
</li>
<li><dl class="first docutils">
<dt>Modify the <cite>samase.pl</cite> script to point to specific PATHs:</dt>
<dd><p class="first">a. Modify the line below to point to where R is installed.</p>
<div class="highlight-python"><pre>my $RLOC = "/usr/software/R/R-2.12.0/bin/R";</pre>
</div>
<p>b. Modify the line below to point to the directory where this file is located.</p>
<div class="last highlight-python"><pre>my $gencode_data_file = "/usr/resources/gencode.v7.annotation.gtf";</pre>
</div>
</dd>
</dl>
</li>
</ol>
</div>
<div class="section" id="part-1-aligning-expression-data">
<h1><strong>Part 1: Aligning Expression Data</strong><a class="headerlink" href="#part-1-aligning-expression-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="input-files">
<h2>Input files<a class="headerlink" href="#input-files" title="Permalink to this headline">¶</a></h2>
<p>Prior to detecting allele-specific expression, the raw data needs be aligned to the reference genome and requires the following files as input:</p>
<ol class="arabic simple">
<li>.fq file containing the raw mmPCR-seq data. This file can be downloaded from the Gene Expression Omnibus (GSE51769).</li>
<li>.fa file containing the human reference genome.</li>
</ol>
</div>
<div class="section" id="align-fastq-files-using-star">
<h2><strong>1. Align fastq files using STAR</strong><a class="headerlink" href="#align-fastq-files-using-star" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will align the fastq files downloaded from GEO to the human reference genome.</p>
<p><strong>Input:</strong> .fq file</p>
<p><strong>Output:</strong> .sam file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>STAR ---genomeDir [GENOME_DIR] --readFilesIn [FASTQ_FILE]  --runThreadN [N] --genomeLoad LoadAndKeep  --outFileNamePrefix [OUTPUT_PREFIX]. --outFilterScoreMin [N] --outFilterScoreMinOverLread [N] --outFilterMatchNmin [N] -- outFilterMatchNminOverLread [N]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>STAR --genomeDir genomes/hg19_STAR_junctions --readFilesIn Cerebr-rep1_S2_L001_R1_001.fastq  --runThreadN 4 --genomeLoad LoadAndKeep  --outFileNamePrefix index1.R1. --outFilterScoreMin 0 --outFilterScoreMinOverLread 0 --outFilterMatchNmin 0 -- outFilterMatchNminOverLread 0</pre>
</div>
</div></blockquote>
</div>
<div class="section" id="convert-format-of-expression-file">
<h2><strong>2. Convert format of expression file</strong><a class="headerlink" href="#convert-format-of-expression-file" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will convert the aligned .sam file to a .bam file, which is necessary for ASE detection later.</p>
<p><strong>Input:</strong> .sam file</p>
<p><strong>Output:</strong> .bam file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>samtools view -S -b -o [OUTPUT_FILE] [INPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>samtools view -S -b -o index1.R1.bam index1.R1.sam</pre>
</div>
</div></blockquote>
</div>
</div>
<div class="section" id="part-2-detecting-allele-specific-expression">
<h1><strong>Part 2: Detecting Allele-Specific Expression</strong><a class="headerlink" href="#part-2-detecting-allele-specific-expression" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id1">
<h2>Input files<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>To detect allele-specific expression, the following files are required as input:</p>
<ol class="arabic simple">
<li>.bed or .vcf file containing list of heterozygous sites to call ASE.</li>
<li>.bam file containing the aligned RNA-seq data.</li>
<li>.fa file containing the human reference genome.</li>
</ol>
</div>
<div class="section" id="sort-alignments-using-samtools">
<h2><strong>1. Sort alignments using SAMtools</strong><a class="headerlink" href="#sort-alignments-using-samtools" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will sort the aligned .bam files by the leftmost coordinate positions.</p>
<p><strong>Input:</strong> Aligned .bam file</p>
<p><strong>Output:</strong> Sorted .bam file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>samtools sort [INPUT_FILE] [OUTPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>samtools sort index1.bam  index1.sorted</pre>
</div>
</div></blockquote>
</div>
<div class="section" id="call-variants-using-samtools">
<h2><strong>2. Call variants using SAMtools</strong><a class="headerlink" href="#call-variants-using-samtools" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will call variants in the RNA-Seq data by transposing the aligned data for the heterozygous positions specified for the individual.</p>
<p><strong>Input:</strong> Sorted .bam file, .vcf file, reference FASTA file</p>
<p><strong>Output:</strong> Pileup file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>samtools mpileup -s -B -f [REF_FILE] [INPUT_FILE] -l [VCF_FILE] &gt; [OUTPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>samtools mpileup -B -f hg19.fa index1.sorted.bam -l individual.hets.vcf &gt; index1.pileup.txt</pre>
</div>
<dl class="docutils">
<dt><strong>Options:</strong></dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-B</span></kbd></td>
<td>Disable probabilistic realignment for the computation of base alignment quality</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-l</span></kbd></td>
<td>Position list file containing a list of sites where pileup should be generated</td></tr>
</tbody>
</table>
</dd>
</dl>
</div></blockquote>
</div>
<div class="section" id="parse-pileup-file">
<h2><strong>3. Parse pileup file</strong><a class="headerlink" href="#parse-pileup-file" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will parse the mpileup file into an easily readable format.</p>
<p><strong>Input:</strong> Pileup file, .vcf file</p>
<p><strong>Output:</strong> Parsed pileup file, .vcf file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>perl samase.pl  -parse_pileup --sp [INPUT_FILE] –vcf [VCF_FILE] &gt; [OUTPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>perl samase.pl  -parse_pileup --sp index1.pileup.txt –vcf individual.hets.vcf &gt; index1.parsed_pileup.txt</pre>
</div>
</div></blockquote>
</div>
<div class="section" id="create-gene-snp-mapping-file">
<h2><strong>4. Create gene-snp mapping file</strong><a class="headerlink" href="#create-gene-snp-mapping-file" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will map each variant to an annotated gene. This is useful so you know which genes the SNPs are testing for ASE later. The location of the annotation file (e.g. gencode.v16.gtf) needs to be indicated the samase.pl script.</p>
<p><strong>Input:</strong> .vcf file</p>
<p><strong>Output:</strong> Gene/snp mapping file</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>perl samase.pl  --get_gene_snp_mapping_file --use_gencode -is [VCF_FILE] &gt; [OUTPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>perl samase.pl  --get_gene_snp_mapping_file --use_gencode -is individual.hets.vcf &gt; gene_snp_mapping_file.txt</pre>
</div>
</div></blockquote>
</div>
<div class="section" id="binomial-test-for-ase">
<h2><strong>5. Binomial test for ASE</strong><a class="headerlink" href="#binomial-test-for-ase" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p><strong>Description:</strong> This command will conduct a binomial test using the SNP positions specified in the .vcf file.</p>
<p><strong>Input:</strong> .vcf file, parsed pileup file, gene/snp mapping file</p>
<p><strong>Output:</strong> ASE binomial test file. The output files will be equivalent to the processed files that can be downloaded from the Gene Expression Omnibus (GSE51769).</p>
<p><strong>Command:</strong></p>
<div class="highlight-python"><pre>perl samase.pl --calculate_ase_normal --pp [PARSED_PILEUP_FILE] –is [VCF_FILE] -r 0.5 –gf [GENE_SNP_MAPPING_FILE] &gt; [OUTPUT_FILE]</pre>
</div>
<p><strong>Example:</strong></p>
<div class="highlight-python"><pre>perl samase.pl --calculate_ase_normal --pp index1.parsed_pileup.txt –is individual.hets.vcf  -r 0.56 –gf gene_snp_mapping_file.txt -mrs 20 -bq 10 &gt; index1.ase.txt</pre>
</div>
<dl class="docutils">
<dt><strong>Options:</strong></dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-r</span></kbd></td>
<td>Expected reference to non-reference allele mapping ratio</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-m<var>rs</var></span></kbd></td>
<td>Minimum number of reads before a test for ASE will be performed</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-b<var>q</var></span></kbd></td>
<td>Minimum base quality tolerated</td></tr>
</tbody>
</table>
</dd>
</dl>
</div></blockquote>
</div>
</div>
<div class="section" id="additional-resources">
<h1>Additional Resources<a class="headerlink" href="#additional-resources" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="https://code.google.com/p/rna-star/">STAR manual</a></p>
<p><a class="reference external" href="http://samtools.sourceforge.net/">SAMtools manual</a></p>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="#">ASE 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Kimberly Kukurba.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b3.
    </div>
  <script type="text/javascript" src="../../js/bundle.js"></script></body>
</html>